hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -531,6 +533,10 @@ private NonSelectQueryPlan buildNonSelectQueryPlan() {
 			return buildUpdateQueryPlan();
 		}
 
+		if ( getSqmStatement() instanceof SqmInsertStatement ) {
+			return buildInsertQueryPlan();
+		}
+
 		throw new NotYetImplementedException( "Query#executeUpdate for Statements of type [" + getSqmStatement() + "not yet supported" );
```

Comment:
Just noticed the above exception message lacks ']' somewhere, though outside the scope of this PR.